### DISCUSSION TITLE
How to access environmental variables in template.html file?

### DISCUSSION THREAD

**Discussion by micahjon** [preactjs/preact-cli#1703] [Q&A] [answered]
I have environmental variables working in my JS code, but wondered if there was a way to access them in the `template.html`

Ideally, I could update an object in `preact.config.js` and then access it's properties in `template.html` with EJS syntax. I'm guessing this is possible--I'm just not very familiar with the Webpack magic that's happening behind the scenes. 

Thanks!

**Comment by rschristian** [comment, marked as answer]
Yep, you sure can!

```js
// preact.config.js
export default (config, env, helpers) => {
    const { plugin: htmlWebpackPlugin } = helpers.getPluginsByName(config, 'HtmlWebpackPlugin')[0];
    htmlWebpackPlugin.options.config.foobarbaz = true;
};
```

```html
<!-- src/template.html -->
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <title><% preact.title %></title>
        <meta name="viewport" content="width=device-width,initial-scale=1">
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <link rel="apple-touch-icon" href="/assets/icons/apple-touch-icon.png">
        <% if (htmlWebpackPlugin.options.config.foobarbaz) { %>
            <meta name="example-meta" content="Hello World!">
        <% } %>
        <% preact.headEnd %>
    </head>
    <body>
        <% preact.bodyEnd %>
    </body>
</html>
```

`config` within `html-webpack-plugin`'s options is the same as `env` in your `preact.config.js` -- it just carries a bunch of standard environment info. You can add additional entries there (as I do above with `foobarbaz`) or add another value into `html-webpack-plugin`'s options, either works just as well. Just need to pass your env var into the options and then EJS will process any conditionals or what have you.

`process` itself won't be defined when EJS runs which is why you'll need to pass them in to `html-webpack-plugin`'s options.
